### PR TITLE
Day5: First SpringBoot with Hackerrank and Attendance

### PR DESCRIPTION
**Pull request by Ghumber**

### CODE REVIEW

File: Humbertino_Avilez/Day5/First-spring-bootcamp/src/main/java/com/bolsadeideas/springboot/bootcamp/app/services/Implements/JavaPatternSyntaxCheckerImpl.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+public class JavaPatternSyntaxCheckerImpl implements JavaPatternSyntaxChecker {
+	private static final Logger javaLog = Logger.getLogger(JavaPatternSyntaxCheckerImpl.class);
+
+	@Override
+	public String validatePatternRedex(String patternToValidate) {
+		try {
+			Pattern.compile(patternToValidate);
+			javaLog.debug(ANSWER_VALID);
+			return ANSWER_VALID;
+		} catch (Exception exception) {
```
Comment (luisod78) [inline Humbertino_Avilez/Day5/First-spring-bootcamp/src/main/java/com/bolsadeideas/springboot/bootcamp/app/services/Implements/JavaPatternSyntaxCheckerImpl.java:25]: Prefer Specific Exceptions
https://dzone.com/articles/9-best-practices-to-handle-exceptions-in-java

File: Humbertino_Avilez/Day5/First-spring-bootcamp/src/main/java/com/bolsadeideas/springboot/bootcamp/app/utils/Constans.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package com.bolsadeideas.springboot.bootcamp.app.utils;
+
+public class Constans {
```
Comment (luisod78) [inline Humbertino_Avilez/Day5/First-spring-bootcamp/src/main/java/com/bolsadeideas/springboot/bootcamp/app/utils/Constans.java:3]: typo: Constants

File: Humbertino_Avilez/Day5/First-spring-bootcamp/src/main/java/com/bolsadeideas/springboot/bootcamp/app/services/JavaAnagramsService.java [hunk 3]
```diff
@@ -0,0 +1,6 @@
+package com.bolsadeideas.springboot.bootcamp.app.services;
+
+public interface JavaAnagramsService {
+	
+	public String isAnagram(String word, String anagram);
```
Comment (luisod78) [inline Humbertino_Avilez/Day5/First-spring-bootcamp/src/main/java/com/bolsadeideas/springboot/bootcamp/app/services/JavaAnagramsService.java:5]: In Java when you see a method with prefix "is" means that method return a boolean.

**Review by luisod78 — CHANGES_REQUESTED** [review]

File: Humbertino_Avilez/Day5/First-spring-bootcamp/src/main/java/com/bolsadeideas/springboot/bootcamp/app/controllers/HackerrankController.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+import com.bolsadeideas.springboot.bootcamp.app.services.JavaAnagramsService;
+import com.bolsadeideas.springboot.bootcamp.app.services.JavaPatternSyntaxChecker;
+import com.bolsadeideas.springboot.bootcamp.app.services.JavaStringReverseService;
+
+@RestController
+public class HackerrankController {
+
+	@Autowired
+	JavaStringReverseService javaStringReverseService;
```
Comment (luisod78) [inline Humbertino_Avilez/Day5/First-spring-bootcamp/src/main/java/com/bolsadeideas/springboot/bootcamp/app/controllers/HackerrankController.java:16]: no access modifiers ?

File: Humbertino_Avilez/Day5/First-spring-bootcamp/src/main/java/com/bolsadeideas/springboot/bootcamp/app/services/Implements/JavaAnagramsServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+	Map<Character, Integer> anagramMap;
+
+	@Override
+	public String validateAnagram(String word, String anagram) {
+		wordMap = new Hashtable<>();
+		anagramMap = new Hashtable<>();
+		anagram = anagram.toLowerCase();
+		if (word.length() == anagram.length()) {
+			for (int i = 0; i < word.length(); i++) {
+				if (wordMap.containsKey(word.charAt(i))) {
```
Comment (luisod78) [inline Humbertino_Avilez/Day5/First-spring-bootcamp/src/main/java/com/bolsadeideas/springboot/bootcamp/app/services/Implements/JavaAnagramsServiceImpl.java:25]: There are 2 if-blocks almost identical... can you see??
only difference is the map is using... so, why don't declare a private method that receives a map and then.... 
follow me????

**Review by luisod78 — CHANGES_REQUESTED** [review]

File: Humbertino_Avilez/Day5/First-spring-bootcamp/src/main/java/com/bolsadeideas/springboot/bootcamp/app/utils/Constans.java (hunk 2 above)
Comment (luisod78) [reply Humbertino_Avilez/Day5/First-spring-bootcamp/src/main/java/com/bolsadeideas/springboot/bootcamp/app/utils/Constans.java]: you missed this comment

File: Humbertino_Avilez/Day5/First-spring-bootcamp/src/main/java/com/bolsadeideas/springboot/bootcamp/app/services/JavaAnagramsService.java (hunk 3 above)
Comment (luisod78) [reply Humbertino_Avilez/Day5/First-spring-bootcamp/src/main/java/com/bolsadeideas/springboot/bootcamp/app/services/JavaAnagramsService.java]: so??? nothing to say? did u miss the comment?

File: Humbertino_Avilez/Day5/First-spring-bootcamp/src/main/java/com/bolsadeideas/springboot/bootcamp/app/controllers/HackerrankController.java (hunk 4 above)
Comment (luisod78) [reply Humbertino_Avilez/Day5/First-spring-bootcamp/src/main/java/com/bolsadeideas/springboot/bootcamp/app/controllers/HackerrankController.java]: missed this comment

**Review by luisod78 — APPROVED** [review]

**Review by globant-bootcamp — APPROVED** [review]

**Pull request merged**